### PR TITLE
ref(stacktrace-link): Transaction tags to stacktrace linking 

### PR DESCRIPTION
**Pull request by MeredithAnya**
**Sentry Transactions:**
I've added a handful of tags to the transaction for the `project_stacktrace_link` endpoint:

`stacktrace_link.found`
`stacktrace_link.error`
`stacktrace_link.platform`
`stacktrace_link.tried_version`
`stacktrace_link.used_version`
`integration_provider` (this could be used across features)

Here is what an example search might look like: 

Screen Shot 2020-12-04 at 12 19 56 PM

**Amplitude Analytics**
I also added a few more amplitude analytics, so that we can track when they start the setup, and whether or not they chose the automatic or manual routes. If the automatic route worked, we should be able to see the `'integrations.stacktrace_link_clicked'` next, and if it didn't I would expect `'integrations.reconfigure_stacktrace_setup'`.

```python 
'integrations.stacktrace_start_setup'
'integrations.stacktrace_automatic_setup'
'integrations.stacktrace_manual_setup'
```

### CODE REVIEW

File: src/sentry/api/endpoints/project_stacktrace_link.py [hunk 1]
```diff
@@ -53,24 +53,33 @@ def get(self, request, project):
...
-            # if we found a match, we can break
-            break
+        with configure_scope() as scope:
+            for config in configs:
+
+                result["config"] = serialize(config, request.user)
+                # use the provider key to be able to spilt up stacktrace
+                # link metrics by integration type
+                provider = result["config"]["provider"]["key"]
+                scope.set_tag("integration", provider)
```
Comment (scefali) [inline src/sentry/api/endpoints/project_stacktrace_link.py:63]: @MeredithAnya can we call this `integration_provider`?

**Review by scefali — COMMENTED** [review]
@MeredithAnya can we add transaction tags for whether using the release (if provided) worked and whether we fall back to the default?

File: src/sentry/integrations/repositories.py [hunk 2]
```diff
@@ -49,12 +50,16 @@ def get_stacktrace_link(self, repo, filepath, default, version):
...
-        if version:
-            source_url = self.check_file(repo, filepath, version)
-            if source_url:
-                return source_url
-
-        source_url = self.check_file(repo, filepath, default)
+        with configure_scope() as scope:
+            scope.set_tag("stacktrace_link.tried_version", False)
+            if version:
+                scope.set_tag("stacktrace_link.tried_version", True)
```
Comment (scefali) [inline src/sentry/integrations/repositories.py:56]: @MeredithAnya are you sure this will wipe out the value set two lines above?

Comment (MeredithAnya) [reply src/sentry/integrations/repositories.py:56]: @scefali testing this out locally it did replace the previously set tag value

**Review by scefali — APPROVED** [review]

**Pull request merged**